Add plugin: Table Checkbox Renderer

dannns opened the pull request:
Add plugin: Table Checkbox Renderer

# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/dannns/obsidian-table-checkbox-renderer

## Release Checklist
- [ ] I have tested the plugin on
  - [ ]  Windows
  - [ ]  macOS
  - [x]  Linux
  - [x]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

dannns commented:
Updated authorUrl to point to my Github profile

joethei commented:
Remove the `main.js` file from the repo and add it to the `.gitignore`, release assets should not be mixed with source assets.

[# Obsidian Table Checkbox Renderer](https://github.com/dannns/obsidian-table-checkbox-renderer/blob/e44b4d31cc9254dd99535c00202aa5cd767fa37c/README.md#L1)
Naming something "Obsidian xyz" is reserved for first party products we create, please change this header.

[const content = await plugin.app.vault.read(file);](https://github.com/dannns/obsidian-table-checkbox-renderer/blob/e44b4d31cc9254dd99535c00202aa5cd767fa37c/src/dom-helpers.ts#L32-L42)
It would be safer to use the `process` function from the API, as that will make sure that the change happens atomically.

dannns commented:
> Remove the `main.js` file from the repo and add it to the `.gitignore`, release assets should not be mixed with source assets.
> 
> [# Obsidian Table Checkbox Renderer](https://github.com/dannns/obsidian-table-checkbox-renderer/blob/e44b4d31cc9254dd99535c00202aa5cd767fa37c/README.md#L1) Naming something "Obsidian xyz" is reserved for first party products we create, please change this header.
> 
> [const content = await plugin.app.vault.read(file);](https://github.com/dannns/obsidian-table-checkbox-renderer/blob/e44b4d31cc9254dd99535c00202aa5cd767fa37c/src/dom-helpers.ts#L32-L42) It would be safer to use the `process` function from the API, as that will make sure that the change happens atomically.

Thank you. Addressed review comments in:
https://github.com/dannns/obsidian-table-checkbox-renderer/releases/tag/1.0.0
